parlai/agents/torch_agent/torch_agent.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from parlai.core.agents import Agent
+from parlai.core.dict import DictionaryAgent
+
+try:
+    import torch
+except Exception as e:
+    raise ModuleNotFoundError('Need to install Pytorch: go to pytorch.org')
+
+import random
+
+
+class TorchAgent(Agent):
+    """Base Agent for all models which use Torch.
+
+    This agent serves as a common framework for all ParlAI models which want
+    to use PyTorch.
+    """
+
+    @staticmethod
+    def dictionary_class():
+        return DictionaryAgent
+
+    def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+
+        self.use_cuda = not opt['no_cuda'] and torch.cuda.is_available()
+        if self.use_cuda:
+            print('[ Using CUDA ]')
+            torch.cuda.device(opt['gpu'])
+
+        if not shared:
+            # Need to set up the model from scratch
+            self.dict = DictionaryAgent(opt)
+        else:
+            # ... copy initialized data from shared table
+            self.opt = shared['opt']
+            self.dict = shared['dict']
+
+        self.NULL_IDX = self.dict[self.dict.null_token]
+        self.END_IDX = self.dict[self.dict.end_token]
+        self.START_IDX = self.dict[self.dict.start_token]
+
+    def share(self):
+        shared = super().share()
+        shared['opt'] = self.opt
+        shared['dict'] = self.dict
+        return shared
+
+    def vectorize(self, obs, use_cuda=True):
```

Comment:
default cuda on true might not be good, can't it just take this from the parent  self.use_cuda or something? it seems to be there?